### ISSUE TITLE
Material surfaces are fogged incorrectly

### ISSUE THREAD

**Issue by jdolan** [jdolan/quetoo#312]
While adding fog to Torn Glory, I noticed that the computer screens (material animations) appear to be fogged twice (?)

![quetoo043](https://cloud.githubusercontent.com/assets/643118/21573588/516a883c-ceb4-11e6-97eb-08006e6b7b8d.png)

**Comment by Paril** [comment]
So, this seems to be because fog is applied to every rendered material that isn't a pulsing object.. I'm not so sure why/how this worked in the past, because these materials are alpha blended and will naturally blend the fogged output pixels with the ones behind it.

The only workaround would be to disable fogging on alpha blended surfaces, but that might cause more issues, I'm not really sure. Thoughts?

**Comment by jdolan** [comment]
I think I just used to disable fog before R_DrawMaterialSurfaces :|

**Comment by Paril** [comment]
I thought so too, but you have it set so that terrain stages, etc get fogged - if we did that, terrain wouldn't get fogged at all.

**Comment by Panjoo** [comment]
Lol, yeah I already came across this during the makings of pits, warehouse and tokays, quite some time ago. It's not just on animated materials. It happens on any 24bit (read: no alpha channel) tga texture with black color that's using _blend GL_SRC_ALPHA GL_ONE_

On warehouse, the blue and yellow lightstrips along the floors would take on the same color as the fog, if I used textures that had black color as background (without alpha). I solved this by making them into transparent png's, but this isn't always easy to do if you only have a 24bit unmasked and flat tga as base. 
I'm in the middle of drawing newer (32bit alpha) tga textures for all of my lamp fx textures. Textures that have an actual alpha mask don't have this problem.

**Comment by jdolan** [comment]
That's super helpful @Panjoo. We should fix the blend ops or convert to RGBA on upload if necessary. Another option, @Paril, which might prove helpful anyway, would be to introduce a dog param to the material definition to give mappers explicit control over how fog is applied to a material.

**Comment by jdolan** [comment]
Fog param. Fucking phone.

**Comment by Paril** [comment]
Interesting.. I'm not so sure why it would appear on those materials but not regularly blended ones. Can you try with a straight colored material with 127 alpha (50% opacity) and see what happens?

**Comment by Paril** [comment]
You can now have stages skip fog with the "skip_fog" flag. Commit 406d640c7e56ed26edbb9bbc876a44bbcf19e681 has added it. I won't close this till we're sure it works though.

**Comment by Panjoo** [comment]
Aha okay, yes that's even better I guess. So if I understand, the only times I have to write that new fog stage is if I DO want a blended tetxure affected by the fog - otherwise nothing has to be changed.

---

BTW, I noticed earlier that the 'glow' blends that I used on all of my lamp/light textures is no longer tranparent. It's very noticable on the jumppad fx on Aero. That blue glass blend was drawn slighty transparent and is now fully opaque. Same goes for all the lamp glow blends in the rest of the maps which became too bright. 
So now that this fog issue has been fixed I'm probably still going to replace most of the png's with tga's and and create alpha's for them with some built-in transparancy to prevent this.
